components/tidb_query_expr/src/impl_string.rs

```diff
@@ -1014,6 +1014,52 @@ pub fn repeat(input: BytesRef, cnt: &Int, writer: BytesWriter) -> Result<BytesGu
     Ok(writer.finish())
 }
 
+#[rpn_fn(writer)]
+#[inline]
+pub fn substring_2_args_utf8(
+    input: BytesRef,
+    pos: &Int,
+    writer: BytesWriter,
+) -> Result<BytesGuard> {
+    substring_utf8(input, *pos, input.chars().count() as Int, writer)
```

Comment:
`chars().count()` iterates the whole string once, and the string is also iterated a few more times. Could you try to optimize it?

Reply:
done.